elasticdl/python/elasticdl/api.py

```diff
@@ -31,39 +35,44 @@ def train(args):
             docker_tlscert=args.docker_tlscert,
             docker_tlskey=args.docker_tlskey,
         )
+    )
 
-        container_args = [
-            "-m",
-            "elasticdl.python.master.main",
-            "--worker_image",
-            image_name,
-            "--model_zoo",
-            _model_zoo_in_docker(model_zoo),
-            "--cluster_spec",
-            _cluster_spec_def_in_docker(args.cluster_spec),
-        ]
-        container_args.extend(
-            build_arguments_from_parsed_result(
-                args, filter_args=["model_zoo", "cluster_spec", "worker_image"]
-            )
+    container_args = [
+        "-m",
+        "elasticdl.python.master.main",
+        "--worker_image",
+        image_name,
+        "--model_zoo",
+        _model_zoo_in_docker(model_zoo),
+        "--cluster_spec",
+        _cluster_spec_def_in_docker(args.cluster_spec),
```

Comment:
Also, if not building image, `cluster_spec` should not set from `_cluster_spec_def_in_docker `.

Reply:
Good suggestion, updated.